### PR TITLE
fix(utils): unwrap may not return previous value

### PR DESCRIPTION
**Pull request by satoren**
## Related Issues or Discussions

Fixes #2214

## Summary

Fixed to carry over previous values

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Review by dai-shi — APPROVED** [review]
I added 3744b21 and modified the description slightly.

File: src/vanilla/utils/unwrap.ts [hunk 1]
```diff
@@ -95,10 +95,10 @@ export function unwrap<Value, Args extends unknown[], Result, PendingValue>(
       return atom(
         (get) => {
           const state = get(promiseAndValueAtom)
-          if ('v' in state) {
-            return state.v
+          if ('f' in state) {
+            return state.f
```
Comment (satoren) [inline src/vanilla/utils/unwrap.ts:99]: It looks a little odd that f(allback) seems to have priority over v(alue).

Comment (dai-shi) [reply src/vanilla/utils/unwrap.ts:99]: Yeah, but with 'f' and 'v', it means 'v' is obsolete.

```js
if ('v' in state && !('f' in state)) {
  return state.v
}
return state.f
```
that requires more bytes.

Comment (dai-shi) [inline src/vanilla/utils/unwrap.ts:99]: We can add a comment.
```suggestion
          if ('f' in state) { // is pending
            return state.f
```

**Pull request merged**